Library/Homebrew/cmd/uses.rb

```diff
@@ -47,49 +47,48 @@ def uses
 
     uses = formulae.select do |f|
       used_formulae.all? do |ff|
-        if recursive
-          deps = f.recursive_dependencies do |dependent, dep|
-            if dep.recommended?
-              Dependency.prune if ignores.include?("recommended?") || dependent.build.without?(dep)
-            elsif dep.optional?
-              Dependency.prune if !includes.include?("optional?") && !dependent.build.with?(dep)
-            elsif dep.build?
-              Dependency.prune unless includes.include?("build?")
+        begin
+          if recursive
+            deps = f.recursive_dependencies do |dependent, dep|
+              if dep.recommended?
+                Dependency.prune if ignores.include?("recommended?") || dependent.build.without?(dep)
+              elsif dep.optional?
+                Dependency.prune if !includes.include?("optional?") && !dependent.build.with?(dep)
+              elsif dep.build?
+                Dependency.prune unless includes.include?("build?")
+              end
             end
-
-            # If a tap isn't installed, we can't find the dependencies of one
-            # its formulae, and an exception will be thrown if we try.
-            if dep.is_a?(TapDependency) && !dep.tap.installed?
-              Dependency.keep_but_prune_recursive_deps
+            reqs = f.recursive_requirements do |dependent, req|
```

Comment:
> If not, I'll add a warning for developers encouraging them to open an issue if they see it. After a while, if nobody reports anything, we could enable the warning for everybody, and then finally if nobody has still reported anything we could remove the rescue.

This sounds good except: test it on all of the tapped Homebrew/* taps, make it reraise for developers rather than warn and then we'll remove it, yeh 👍